packages/stores/src/account/osmosis/index.ts

```diff
@@ -2040,6 +2044,110 @@ export class OsmosisAccountImpl {
     );
   }
 
+  /**
+   * Method to undelegate from validator set.
+   * @param coin The coin object with denom and amount to undelegate.
+   * @param memo Transaction memo.
+   * @param onFulfill Callback to handle tx fulfillment given raw response.
+   */
+  async sendUndelegateFromValidatorSetMsg(
+    coin: CoinType,
+    memo: string = "",
+    onFulfill?: (tx: any) => void
+  ) {
+    await this.base.signAndBroadcast(
+      this.chainId,
+      "undelegateFromValidatorSet",
+      async () => {
+        const msg = this.msgOpts.undelegateFromValidatorSet.messageComposer({
+          delegator: this.address,
+          coin: {
+            denom: coin.denom.coinMinimalDenom,
+            amount: coin.amount,
+          },
+        });
+
+        return [msg];
+      },
```

Comment:
For both functions here, the 3rd param is a list of msg objects

I think it'd be more readable if we remove the async function and directly return the message composer return value wrapped in a array literal

Reply:
I'm not following. Please paste code to demonstrate what you mean.

Reply:
Replace the whole line range with:

```
[
  this.msgOpts.undelegateFromValidatorSet.messageComposer({
    delegator: this.address,
    coin: {
      denom: coin.denom.coinMinimalDenom,
      amount: coin.amount,
    },
  })
]
```

Less cluttered